packages/compose/src/with-safe-interval/index.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * External dependencies
+ */
+import { without } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import createHigherOrderComponent from '../create-higher-order-component';
+
+/**
+ * A higher-order component used to provide and manage interval-based function
+ * calls that ought to be bound to a component's lifecycle.
+ *
+ * @param {Component} OriginalComponent Component requiring setInterval
+ *
+ * @return {Component}                  Wrapped component.
+ */
+const withSafeInterval = createHigherOrderComponent(
+	( OriginalComponent ) => {
+		return class WrappedComponent extends Component {
+			constructor() {
+				super( ...arguments );
+				this.intervals = [];
+				this.setInterval = this.setInterval.bind( this );
+				this.clearInterval = this.clearInterval.bind( this );
+			}
+
+			componentWillUnmount() {
+				this.intervals.forEach( clearInterval );
+			}
+
+			setInterval( fn, delay ) {
+				const id = setInterval( () => {
+					fn();
+					this.clearInterval( id );
```

Comment:
It's not clear to me: Why would the default behavior clear the interval after the initial tick?

Reply:
Good catch! It shouldn't. This was a left-over from the `withSafeTimeout` file, which I copied and then adjusted. 👍